common/tests/integration_tests.rs

```diff
@@ -128,12 +129,201 @@ fn basic2() {
         .collect::<Vec<_>>();
     csv.verify_last_header_finalization(&fp).unwrap();
     light_client.update(block_header, fp).unwrap();
+    let commits = csv.get_total_commits();
     let merkle_tree = OneshotMerkleTree::create(
-        csv.get_total_commits()[1..=3]
+        commits[1..=(commits.len() - 2)]
             .iter()
             .map(|c| c.to_hash256())
             .collect(),
     );
     let merkle_proof = merkle_tree.create_merkle_proof(tx.to_hash256()).unwrap();
     assert!(light_client.verify_transaction_commitment(&tx, 1, merkle_proof));
 }
+
+#[test]
+fn basic3() {
+    let member_number = 10;
+    let (rs, keys) = test_utils::generate_standard_genesis(member_number);
+    let genesis_info = rs.genesis_info.clone();
+    let genesis_header = rs.genesis_info.header.clone();
+
+    let (height, timestamp) = (1, 0);
+    let mut csv = CommitSequenceVerifier::new(genesis_header.clone(), rs.clone()).unwrap();
+    let mut light_client = LightClient::new(genesis_header);
+    let author = &keys[1];
+
+    let tx = Transaction {
+        author: "doesn't matter".to_owned(),
+        timestamp,
+        head: "commit 1".to_owned(),
+        body: "".to_owned(),
+        diff: Diff::None,
+    };
+    csv.apply_commit(&Commit::Transaction(tx.clone())).unwrap();
+
+    let agenda = Agenda {
+        height,
+        author: rs.query_name(&author.0).unwrap(),
+        timestamp,
+        transactions_hash: Agenda::calculate_transactions_hash(&[tx.clone()]),
+    };
+    csv.apply_commit(&Commit::Agenda(agenda.clone())).unwrap();
+
+    csv.apply_commit(&Commit::AgendaProof(AgendaProof {
+        height,
+        agenda_hash: agenda.to_hash256(),
+        proof: keys[1..]
+            .iter()
+            .map(|(_, private_key)| TypedSignature::sign(&agenda, private_key).unwrap())
+            .collect::<Vec<_>>(),
+        timestamp,
+    }))
+    .unwrap();
+    let data = DelegationTransactionData {
+        delegator: rs.members[0].name.to_owned(),
+        delegatee: rs.members[2].name.to_owned(),
+        governance: true,
+        block_height: height,
+        timestamp,
+        chain_name: "pdao".to_owned(),
```

Comment:
```suggestion
        chain_name: "PDAO-mainnet".to_owned(),
```
It's test, but how about using the name we really use?

Reply:
fixed